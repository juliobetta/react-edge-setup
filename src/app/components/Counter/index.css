:root {
  --background: #ededeb;
}

.Counter {
  background-color: var(--background);
  padding: 1rem;
  display: flex;
  justify-content: space-between;
  align-items: center;

  > span {
    font-size: 4rem;
  }
}
