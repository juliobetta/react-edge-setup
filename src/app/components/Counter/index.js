import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, IconButton } from 'react-toolbox/lib/button';
import { initialState, increment, decrement, reset } from './state';
import styles from './index.css';


class Counter extends Component {
  state = initialState(this.props.initialValue);

  render() {
    const { initialValue, step } = this.props;

    return (
      <div className={styles.Counter}>
        <div>
          <Button
            floating
            mini
            onClick={this.handleDecrement(initialValue, step)}
          >
            â€“
          </Button> { ' ' }
          <Button
            floating
            onClick={this.handleIncrement(step)}
          >
            +
          </Button> { ' ' }
          <IconButton icon="replay" id="reset" onClick={this.handleReset(initialValue)} />
        </div>

        <span id="counter">{this.state.counter}</span>
      </div>
    );
  }

  handleIncrement = step => () => this.setState(state => increment(state, step));
  handleDecrement = (initialValue, step) => () => {
    const newState = decrement(this.state, step);

    if(initialValue <= newState.counter) {
      this.setState({ ...this.state, ...newState });
    }
  };
  handleReset = initialValue => () => this.setState(() => reset(initialValue))
}


Counter.propTypes = {
  initialValue: PropTypes.number,
  step: PropTypes.number
};

Counter.defaultProps = {
  initialValue: 1,
  step: 1
};


export default Counter;
