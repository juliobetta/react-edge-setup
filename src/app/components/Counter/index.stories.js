import React from 'react';
import { storiesOf } from '@storybook/react';
import Counter from './index';

storiesOf('Counter  ', module)
  .add('with defaults', () => <Counter />)
  .add('with custom start', () => (
    <Counter initialValue={20} />
  ))
  .add('with custom step', () => (
    <div>
      <Counter initialValue={0} step={10} />
    </div>
  ));
