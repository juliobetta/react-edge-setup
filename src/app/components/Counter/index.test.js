import React from 'react';
import Counter from './index';

describe('<Counter>', () => {
  it('renders correctly', () => {
    const wrapper = mount(<Counter />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
