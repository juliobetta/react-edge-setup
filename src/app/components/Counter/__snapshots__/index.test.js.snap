// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter> renders correctly 1`] = `
<Counter
  initialValue={1}
  step={1}
>
  <div
    className="Counter"
  >
    <div>
      <ThemedThemed
        composeTheme="deeply"
        disabled={false}
        floating={true}
        mapThemrProps={[Function]}
        mini={true}
        onClick={[Function]}
        ripple={true}
        rippleCentered={false}
        rippleClassName=""
        rippleMultiple={true}
        rippleSpread={2}
      >
        <ThemedRippledComponent
          composeTheme="deeply"
          disabled={false}
          floating={true}
          mapThemrProps={[Function]}
          mini={true}
          onClick={[Function]}
          ripple={true}
          rippleCentered={false}
          rippleClassName=""
          rippleMultiple={true}
          rippleSpread={2}
          theme={Object {}}
        >
          <RippledComponent
            disabled={false}
            floating={true}
            mini={true}
            onClick={[Function]}
            ripple={true}
            rippleCentered={false}
            rippleClassName=""
            rippleMultiple={true}
            rippleSpread={2}
            theme={Object {}}
          >
            <Button
              accent={false}
              className=""
              disabled={false}
              flat={false}
              floating={true}
              mini={true}
              neutral={true}
              onClick={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              primary={false}
              raised={false}
              theme={Object {}}
              type="button"
            >
              <button
                className=""
                data-react-toolbox="button"
                disabled={false}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchStart={[Function]}
                type="button"
              >
                â€“
              </button>
            </Button>
          </RippledComponent>
        </ThemedRippledComponent>
      </ThemedThemed>
       
       
      <ThemedThemed
        composeTheme="deeply"
        disabled={false}
        floating={true}
        mapThemrProps={[Function]}
        onClick={[Function]}
        ripple={true}
        rippleCentered={false}
        rippleClassName=""
        rippleMultiple={true}
        rippleSpread={2}
      >
        <ThemedRippledComponent
          composeTheme="deeply"
          disabled={false}
          floating={true}
          mapThemrProps={[Function]}
          onClick={[Function]}
          ripple={true}
          rippleCentered={false}
          rippleClassName=""
          rippleMultiple={true}
          rippleSpread={2}
          theme={Object {}}
        >
          <RippledComponent
            disabled={false}
            floating={true}
            onClick={[Function]}
            ripple={true}
            rippleCentered={false}
            rippleClassName=""
            rippleMultiple={true}
            rippleSpread={2}
            theme={Object {}}
          >
            <Button
              accent={false}
              className=""
              disabled={false}
              flat={false}
              floating={true}
              mini={false}
              neutral={true}
              onClick={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              primary={false}
              raised={false}
              theme={Object {}}
              type="button"
            >
              <button
                className=""
                data-react-toolbox="button"
                disabled={false}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchStart={[Function]}
                type="button"
              >
                +
              </button>
            </Button>
          </RippledComponent>
        </ThemedRippledComponent>
      </ThemedThemed>
       
       
      <ThemedThemed
        composeTheme="deeply"
        disabled={false}
        icon="replay"
        id="reset"
        mapThemrProps={[Function]}
        onClick={[Function]}
        ripple={true}
        rippleCentered={true}
        rippleClassName=""
        rippleMultiple={true}
        rippleSpread={2}
      >
        <ThemedRippledComponent
          composeTheme="deeply"
          disabled={false}
          icon="replay"
          id="reset"
          mapThemrProps={[Function]}
          onClick={[Function]}
          ripple={true}
          rippleCentered={true}
          rippleClassName=""
          rippleMultiple={true}
          rippleSpread={2}
          theme={Object {}}
        >
          <RippledComponent
            disabled={false}
            icon="replay"
            id="reset"
            onClick={[Function]}
            ripple={true}
            rippleCentered={true}
            rippleClassName=""
            rippleMultiple={true}
            rippleSpread={2}
            theme={Object {}}
          >
            <IconButton
              accent={false}
              className=""
              disabled={false}
              icon="replay"
              id="reset"
              neutral={true}
              onClick={[Function]}
              onMouseDown={[Function]}
              onTouchStart={[Function]}
              primary={false}
              theme={Object {}}
              type="button"
            >
              <button
                className=""
                data-react-toolbox="button"
                disabled={false}
                id="reset"
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchStart={[Function]}
                type="button"
              >
                <FontIcon
                  alt=""
                  className=""
                  value="replay"
                >
                  <span
                    aria-label=""
                    className="material-icons"
                    data-react-toolbox="font-icon"
                  >
                    replay
                  </span>
                </FontIcon>
              </button>
            </IconButton>
          </RippledComponent>
        </ThemedRippledComponent>
      </ThemedThemed>
    </div>
    <span
      id="counter"
    >
      1
    </span>
  </div>
</Counter>
`;
